tests/compute/array-param.slang

```diff
@@ -1,10 +1,12 @@
+//TEST(compute):COMPARE_COMPUTE_EX:-cpu -compute
 //TEST(compute):COMPARE_COMPUTE_EX:-slang -compute
 //TEST(compute):COMPARE_COMPUTE_EX:-slang -compute -dx12
 //TEST(compute, vulkan):COMPARE_COMPUTE_EX:-vk -compute
 
-//TEST_INPUT:ubuffer(data=[0 0 0 0], stride=4):dxbinding(0),glbinding(0),out
 
+//TEST_INPUT:ubuffer(data=[0 0 0 0], stride=4):dxbinding(0),glbinding(0),out,name outputBuffer
```

Comment:
Aesthetically it would be nice if this was `name=outputBuffer` or `name(outputBuffer)`, but I don't think it matters a whole bunch.

The approach that would match the other back-ends more directly would be to use `cpuoffset(...)` to specify the byte offset of the parameter in the uniform-data structure. I think I like your approach better, since we've always said we should be using reflection to set this stuff up where possible.

It would be nice if we have an issue to track the eventual cleanup task of going through and making a reflection-based approach to binding these parameters that works across *all* the renderer back-ends (at least when using Slang). We might still need to keep the non-reflection-based option when doing comparisons against raw HLSL/GLSL shaders, but those test cases are only a small fraction of the total these days.